umami/gp.py

```diff
@@ -56,14 +55,11 @@ def dist2(ls, x1, x2=None):
 def grad_dist2(ls, x1, x2=None):
     if x2 is None:
         x2 = x1
-        
+
     # Rescale.
     x1 = x1 / ls
     x2 = x2 / ls
-    
-    N = x1.shape[0]
```

Comment:
Fixed the mistake and squashed the commit.

Unfortunately pyflakes doesn't honor `# noqa`, so I'm now using flake8. It should be configurable to only check for certain errors in a build script, until we get them all ironed out.